### PR TITLE
Protect important directories

### PR DESCRIPTION
**Pull request by mvoevodskiy**
### What does it do?
Added list of protected directories, that can not be renamed or removed from manager:

- MODX_ASSETS_PATH
- MODX_BASE_PATH
- MODX_CONNECTORS_PATH
- MODX_CORE_PATH
- MODX_MANAGER_PATH
- MODX_PROCESSORS_PATH
- XPDO_CORE_PATH

Renaming or removing this directories will cause error with message "Access Denied".

Also, this PR contain 2 small fixes in processor browser/directory/rename

### Why is it needed?
Description from issue:
`Right click on any of these folders, rename or delete them and you are in trouble.
Could also reduce impact of malicious actions by people with low morals or a short temper.`

### Related issue(s)/PR(s)
#14114

### CODE REVIEW

**Comment by JoshuaLuckers** [comment]
Thanks for taking time to create this PR and help make MODX better. It seems that you are a first-time contributor. To prevent us from merging this PR it's important that you sign the [Contributor License Agreement (CLA)](https://docs.modx.com/community/contribute/becoming-a-contributor).

If you have signed the CLA already please let us know. If you have any questions feel free to post them.

**Comment by mvoevodskiy** [comment]
@JoshuaLuckers thanks for link to CLA. I have already signed today.

**Review by alroniks — CHANGES_REQUESTED** [review]
I would love to see a more descriptive error message with an explanation of why user can to do such action.

**Comment by mvoevodskiy** [comment]
> I would love to see a more descriptive error message with an explanation of why user can to do such action.

Added.

**Review by JoshuaLuckers — COMMENTED** [review]
Please only include the English lexicon entries. Lexicons in other languages are managed via CrowdIn and they might get overwritten when translations are imported.

File: core/lexicon/ru/file.inc.php [hunk 1]
```diff
@@ -61,7 +61,9 @@
 $_lang['file_folder_err_perms_remove'] = 'У веб-сервера не хватает прав для удаления каталога.';
 $_lang['file_folder_err_perms_upload'] = 'У веб-сервера не хватает прав для загрузки в каталог.';
 $_lang['file_folder_err_rename'] = 'Возникла неизвестная ошибка при переименовании каталога.';
+$_lang['file_folder_err_rename_protected'] = 'Переименование защищенного системного каталога не допускается.';
```
Comment (alroniks) [inline core/lexicon/ru/file.inc.php]: All translations should be done via Crowdin, so, please, remove Russian strings from the pull request.

**Review by alroniks — CHANGES_REQUESTED** [review]

File: core/model/modx/processors/browser/directory/rename.class.php [hunk 2]
```diff
@@ -11,8 +11,8 @@
 /**
  * Renames a file
```
Comment (Jako) [inline core/model/modx/processors/browser/directory/rename.class.php]: one 'file' is left

Comment (mvoevodskiy) [reply core/model/modx/processors/browser/directory/rename.class.php]: Fixed

**Review by JoshuaLuckers — APPROVED** [review]

**Review by Jako — APPROVED** [review]

**Review by Ruslan-Aleev — APPROVED** [review]
Everything works as described

File: core/model/modx/processors/browser/browser.class.php [hunk 3]
```diff
@@ -76,6 +76,24 @@ public function getSource()
         return $this->source;
     }
 
+    /**
+     * Get the list of protected directories
+     *
+     * @return array
+     */
+    public function getProtectedPathDirectories() {
```
Comment (alroniks) [inline core/model/modx/processors/browser/browser.class.php:84]: Does it make sense to make it static as it returns constants? Not a request, but just a question for discussion.

**Review by alroniks — APPROVED** [review]
The code seems OK for me and works as expected.

**Review by Mark-H — APPROVED** [review]

**Pull request merged**